packages/runtime/src/widgets/Stepper.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useCallback } from "react";
+import {
+  Stepper as MUIStepper,
+  Step,
+  StepLabel,
+  StepButton,
+  Button,
+  Typography,
+  StepOwnProps,
+  styled,
+} from "@mui/material";
+import type { StepIconProps } from "@mui/material/StepIcon";
+import StepConnector, {
+  stepConnectorClasses,
+} from "@mui/material/StepConnector";
+import {
+  type EnsembleWidget,
+  useRegisterBindings,
+} from "@ensembleui/react-framework";
+import type { EnsembleWidgetProps } from "../util/types";
+import { WidgetRegistry } from "../registry";
+import { EnsembleRuntime } from "../runtime";
+
+export interface StepProps {
+  stepLabel: string;
+  activeStepWidget: EnsembleWidget;
+  inactiveStepWidget: EnsembleWidget;
+  contentWidget: EnsembleWidget;
+}
+
+export type StepperProps = {
+  activeStepIndex?: number;
+  steps: StepProps[];
+  styles: {
+    connectorColor?: string;
+    connectorHeight?: number;
+  };
+} & EnsembleWidgetProps;
+
+interface CustomConnectorProps {
+  connectorColor?: string;
+  activeConnectorColor?: string;
+  completedConnectorColor?: string;
+  connectorHeight?: number;
+}
+
+const Stepper: React.FC<StepperProps> = (props) => {
+  const [activeStep, setActiveStep] = useState(0);
+  const handleNext = useCallback(() => {
+    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+  }, [setActiveStep]);
+
+  const handleBack = useCallback(() => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  }, [setActiveStep]);
+  const handleStep = (step: number) => () => {
+    setActiveStep(step);
+  };
+  const { values } = useRegisterBindings(
+    { ...props, activeStep: activeStep },
+    props.id,
+    {
+      handleNext,
+      handleBack,
+    }
+  );
+  return (
+    <div>
+      <MUIStepper
+        activeStep={values?.activeStep}
+        alternativeLabel
+        connector={
+          <CustomConnector
+            connectorColor={props?.styles.connectorColor}
+            connectorHeight={props?.styles.connectorHeight}
+          />
+        }
+      >
+        {values?.steps[0].inactiveStepWidget &&
+        values?.steps[0].activeStepWidget
```

Comment:
Yeah this should be refactored in an item template and rendered like a DataGrid's Cell component